### PR TITLE
Fishingmap/datasets upload

### PR DESCRIPTION
**Pull request by j8seangel**
This PR starts the support of custom datasets in fishing-map with just the UI to create and remove datasets in the API
See design in: https://projects.invisionapp.com/d/main#/console/21003514/442552637/preview with support for editing the name and description too:

https://user-images.githubusercontent.com/10500650/106436404-56317d80-6474-11eb-97b3-2bf0c6bb2fee.mp4

TODO in another PRs to keep this as simple to review as possible:
- [ ] Add dataset from Context Layer Sidebar 
- [ ] Show created datasets as layers
  - [ ] In sidebar
  - [ ] In map using layer-composer
- [ ] Support [advanced datasets](https://projects.invisionapp.com/d/main#/console/21003514/442552640/preview) reading the file and autocomplete the field options

### CODE REVIEW

File: applications/fishing-map/src/data/datasets.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // TODO: rename this to use 4WINGS DATASET TYPE
-export const FISHING_DATASET_TYPE = '4wings:v1'
+export const FOURWINGS_DATASET_TYPE = '4wings:v1'
```
Comment (rdgfuentes) [inline applications/fishing-map/src/data/datasets.ts:2]: Apparently this constant is also defined in our react-hooks https://github.com/GlobalFishingWatch/frontend/blob/fishingmap/datasets-upload/packages/react-hooks/src/use-dataviews-layers/use-dataviews-layers.ts#L12. Should it be renamed too to keep consistency?

File: applications/fishing-map/src/features/datasets/NewDataset.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        className={styles.input}
+        onChange={(e) => onDatasetFieldChange({ name: e.target.value })}
+      />
+      <InputText
+        inputSize="small"
+        label={t('common.description', 'Description')}
+        className={styles.input}
+        onChange={(e) => onDatasetFieldChange({ description: e.target.value })}
+      />
+      {/*
```
Comment (rdgfuentes) [inline applications/fishing-map/src/features/datasets/NewDataset.tsx:40]: I'd normally remove the commented code

Comment (j8seangel) [reply applications/fishing-map/src/features/datasets/NewDataset.tsx:40]: Indeed, but in this case I'm going to need it in the next PR

File: applications/fishing-map/src/data/datasets.ts (hunk 1 above)
Comment (j8seangel) [reply applications/fishing-map/src/data/datasets.ts]: Good point! I moved the DatasetType from type to enum to centralise the value and use it everywhere, see commit: https://github.com/GlobalFishingWatch/frontend/commit/e6dfab02f7bfb6a051fba402835a135b57a10f1c Note: I didn't change Amalthea as it is deprecated and will be removed soon

**Pull request merged**